### PR TITLE
feat: Delete group conversation locally [#WPB-11556]

### PR DESCRIPTION
**Pull request by m-zagorski**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-11556" title="WPB-11556" target="_blank">StoryWPB-11556</a>  [Android] Remove a group from my conversation list after leaving it
  </td></table>
  <br />
 

https://wearezeta.atlassian.net/browse/WPB-11556

# What's new in this PR?

### Issues

_Briefly describe the issue you have solved or implemented with this pull request. If the PR contains multiple issues, use a bullet list._

### Causes (Optional)

_Briefly describe the causes behind the issues. This could be helpful to understand the adopted solutions behind some nasty bugs or complex issues._

### Solutions

_Briefly describe the solutions you have implemented for the issues explained above._

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: app/src/main/kotlin/com/wire/android/ui/calling/ongoing/OngoingCallViewModel.kt [hunk 1]
```diff
@@ -140,14 +138,6 @@ class OngoingCallViewModel @AssistedInject constructor(
         }
     }
 
-    private fun mapQualityStream(uiParticipant: UICallParticipant): CallQuality {
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/calling/ongoing/OngoingCallViewModel.kt]: question: is removing call quality somehow related to deleting group conversation locally? What is the reason for this change? 🤔

Comment (m-zagorski) [reply app/src/main/kotlin/com/wire/android/ui/calling/ongoing/OngoingCallViewModel.kt]: Its not, but it is required to get latest kalium develop

Comment (m-zagorski) [reply app/src/main/kotlin/com/wire/android/ui/calling/ongoing/OngoingCallViewModel.kt]: reverted, on the latest develop the quality is thjere, probably the changes were not having that change at the moment when I created this pr

**Review by saleniuk — APPROVED** [review]

File: app/src/main/kotlin/com/wire/android/workmanager/worker/DeleteConversationLocallyWorker.kt [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import com.wire.kalium.logic.feature.session.CurrentSessionResult
+import com.wire.kalium.logic.functional.fold
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.mapNotNull
+
+@HiltWorker
+class DeleteConversationLocallyWorker @AssistedInject constructor(
```
Comment (yamilmedina) [inline app/src/main/kotlin/com/wire/android/workmanager/worker/DeleteConversationLocallyWorker.kt:54]: Suggestion: can we add some kdocs here, so we have a bit of the idea behind this worker, like: when it runs, who uses it, how to use it, to name a few 🙏

**Review by yamilmedina — COMMENTED** [review]
Looking really good, it would be nice to add some kdocs in the worker, so we have context for the future :D

File: app/src/main/kotlin/com/wire/android/workmanager/worker/DeleteConversationLocallyWorker.kt (hunk 2 above)
Comment (m-zagorski) [reply app/src/main/kotlin/com/wire/android/workmanager/worker/DeleteConversationLocallyWorker.kt:54]: sure ! 👍

**Review by yamilmedina — APPROVED** [review]
👏
